### PR TITLE
Daniel/health displayed

### LINKED ISSUE #50: FR Health Displayed
When a player's character or an enemy character loses health, the character's state should up updated decreasing the health, this health count should also be visually shown in the UI.

### PR DESCRIPTION
**Pull request by DanT52**
Added hearts above characters that correspond to their current health. I updated the Stats.gd script to call the setHearts function in the new hearts container script to keep the number of hearts displayed consistent with the current hp of the character.

image

Currently, if the players' HP drops below 0, it is noted that the character died and the player health is just reset back it the original health of the  character,
resolves #50

### CODE REVIEW

File: battle/hearts_container.gd [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+@onready var stats = $"../../Stats"
+@onready var HeartGuiClass = preload("res://battle/heart_gui.tscn")
+@onready var maxHP = stats.HP
+# Called when the node enters the scene tree for the first time.
+func _ready():
+	setHearts(maxHP) # adds max hp hearts to the character
+
+func setHearts(num: int):
+	var current_hearts = get_child_count()
+	if num > current_hearts: # we have too many hearts, remove the excess
```
Comment (NateShaw2) [inline battle/hearts_container.gd:12]: For the comment for line 12 is it supposed to be that we don't have enough hearts so we need to add more?

**Review by NateShaw2 — COMMENTED** [review]
This PR is almost ready to merge I left a comment regarding line 12 on hearts_container.gd and I'm not sure how you set the enemy hp to be different from the player's hp. Other than that this PR is good to merge

File: battle/hearts_container.gd (hunk 1 above)
Comment (DanT52) [reply battle/hearts_container.gd]: yes, it is I accidentally put that comment twice.

**Comment by DanT52** [comment]
The hp is tied to the stats of the character, so in the hearts container, on ready, we get the parents node's Stats node and set the number of hearts to be what the max hp is in the stats node of that character.

**Pull request merged**